### PR TITLE
cli improvements

### PR DESCRIPTION
**Pull request by cycool29**
Add some `echo` to scripts and replace most `echo` with new api message functions: `status`, `status_green` and `warning`.

Fix some minor bugs.

- Use `\e[0m` as closing for `echo -e` 
- Add few `echo` to create newlines to make output neater
- Replace most `echo` with new api message functions
- Check app name before update in `manage`
- Show error messages if update failed in `manage` 
- Add `reduceapt` in `purge-installed`
- Show `No apps/files is updatable` if there isn't any app/files to update in `updater`

### CODE REVIEW

File: updater [hunk 1]
```diff
@@ -430,50 +434,84 @@ elif [ "$runmode" == cli ];then #return list of updatable apps, and ask the user
...
+    echo
+  else
+    status "These files can be updated:"
+    echo
+    echo "$updatable_files" | sed 's/^/  - /g'
+    echo
+  fi
   
-  read -p "Update now? [Y/n] " answer
+  question=$(status 'Update now? [Y/n] ')
```
Comment (Botspot) [inline updater:464]: Not sure why you're placing the question into an array.

File: manage [hunk 2]
```diff
@@ -196,37 +203,47 @@ elif [ "$1" == 'install' ] || [ "$1" == 'uninstall' ];then
...
-    echo -e "\n\e[42m\e[30m${mode^}ed ${app} successfully.\e[39m\e[49m" | tee -a "$logfile"
+    status_green "\n${mode^}ed ${app} successfully." | tee -a "$logfile"
     
     format_log_file "$logfile" #remove escape sequences from logfile
     mv "$logfile" "$(echo "$logfile" | sed 's+-incomplete-+-success-+g')" #
+    echo
     exit 0
   else
-    echo -e "\n\e[91mFailed to ${mode} ${app} with $scriptname script"\!"\e[39m
+    error "\nFailed to ${mode} ${app} with $scriptname script"\!"\e[39m
```
Comment (Botspot) [inline manage:221]: This breaks the script because the `error ` function runs `exit 1`.

File: manage [hunk 3]
```diff
@@ -275,13 +293,26 @@ elif [ "$1" == 'update' ];then
   cp -rf "${DIRECTORY}/update/pi-apps/apps/${app}" "${DIRECTORY}/apps/${app}"
   
   if [ "$installback" == 'yes' ];then
-    echo "$app was originally installed before updating. Reinstalling the new version now."
```
Comment (Botspot) [inline manage:278]: Why was this line removed?

File: manage [hunk 4]
```diff
@@ -275,13 +293,26 @@ elif [ "$1" == 'update' ];then
...
     "${DIRECTORY}/manage" install "$app"
+    if [ $? == 0 ]; then
+      failed=true
+    else
+      failed=false
+    fi
+  fi
+  
+  if [ "$failed" == 'true' ]; then
+    error "Failed to update ${app}."
```
Comment (Botspot) [inline manage:306]: This breaks the script because the `error ` function runs `exit 1`. In this case, when the first app fails to update, subsequent appd wouldn't be reinstalled at all, because the usage of `error()` would kill the script prematurely.

File: updater [hunk 5]
```diff
@@ -44,9 +44,9 @@ check_update_interval() { #return 0 if update-interval allows update-checking to
   elif [ "$updateinterval" == 'Always' ];then
     return 0
   elif [ -z "$updateinterval" ];then
-    echo "Something isn't right. Does '${DIRECTORY}/data/settings/Check for updates' exist?" 1>&2
+    warning "Something isn't right. Does '${DIRECTORY}/data/settings/Check for updates' exist?" 1>&2
   else
-    echo "Warning: Unrecognized update interval!" 1>&2
+    error "Unrecognized update interval!" 1>&2
```
Comment (Botspot) [inline updater:49]: This breaks the script because the `error ` function runs `exit 1`.

File: updater [hunk 6]
```diff
@@ -217,14 +217,15 @@ update_now_gui() { #input: updatable_files and updatable_apps variables
     mkdir -p "$(dirname "${DIRECTORY}/${file}")"
     
     #copy new version
-    cp -f "${DIRECTORY}/update/pi-apps/${file}" "${DIRECTORY}/${file}" || echo -e "\e[91mFailed to copy ${DIRECTORY}/update/pi-apps/${file}\e[39m!"
+    cp -f "${DIRECTORY}/update/pi-apps/${file}" "${DIRECTORY}/${file}" || error "Failed to copy ${DIRECTORY}/update/pi-apps/${file}!"
```
Comment (Botspot) [inline updater:220]: This breaks the script because the `error ` function runs `exit 1`.

File: updater [hunk 7]
```diff
@@ -258,14 +259,14 @@ update_now_cli() { #input: updatable_files and updatable_apps variables
...
```
Comment (Botspot) [inline updater:262]: This breaks the script because the `error ` function runs `exit 1`.

**Comment by Botspot** [comment]
@cycool29 I've reviewed your PR and made comments as necessary.

File: updater (hunk 1 above)
Comment (cycool29) [reply updater:464]: > Not sure why you're placing the question into an array.

Just want to have `status` colors to the question.

File: manage (hunk 3 above)
Comment (cycool29) [reply manage:278]: There is a line above it mentioned `status "$app's install script has been updated. Reinstalling $app..."`.

File: updater (hunk 1 above)
Comment (Botspot) [reply updater:464]: > > Not sure why you're placing the question into an array.
> 
> Just want to have `status` colors to the question.

Oh I see. This may be fine for now, but down the road I may tweak the `status` function to be more than blue text. It may prefix everything with "`Status: `", or underline the text, or something else undesirable for asking a question. Son in this case, I recommend you replace it with `echo -en` (and then the codes for the blue color) Color codes can be found [here](https://misc.flogisoft.com/bash/tip_colors_and_formatting).

Comment (cycool29) [reply updater:464]: Just found that can be done like this: `read -p "$(status 'Update now? [Y/n] ')" answer`.

**Comment by cycool29** [comment]
> @cycool29 I've reviewed your PR and made comments as necessary.

Thanks! 
I have updated the scripts and fixed some bugs.
